common/interactive-object/server.mjs

```diff
@@ -239,6 +240,25 @@ class InteractiveObjectServerBaseClass {
     return this.sf_exit(playerID, {next: this.dataStore.get(playerID)});
   }
 
+  async sf_textBox(playerID, kwargs) {
+    const {nextState, dialog, key} = kwargs;
+    const res = await this.helper.callS2cAPI(playerID, 'dialog',
+    'showDialogWithPrompt', 60*1000, this.objectName, dialog);
+    if (res.msg === key) return nextState;
+
+    //The key is wrong,
+    return this.sf_exit(playerID, {next: this.dataStore.get(playerID)});
```

Comment:
For maximum flexibility, maybe add a nextStateIncorrect instead of directly doing sf_exit()?